File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* tmp = nullptr;
        ListNode* n = nullptr;
        ListNode* p = nullptr;
        ListNode* c = head;

        while(c){
            tmp = c->next;
            c->next = p;
            p = c;
            c = tmp;
        }
        return p;
    }
};

File: Data Structures & Algorithms/maximum-twin-sum-of-a-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        maxPair = 0
        s,f = head, head
        while f and f.next:
            f = f.next.next
            s = s.next
        mid = s
        prev = None
        while mid:
            n = mid.next
            mid.next = prev
            prev = mid
            mid = n
        mid = prev
        s = head
        while mid:
            maxPair = max(maxPair, mid.val + s.val)
            mid = mid.next
            s = s.next
        return maxPair